core/src/macosX64Main/kotlin/CentralManagerDelegate.kt

```diff
@@ -0,0 +1,136 @@
+
+    // todo: MutableSharedFlow
+    private val _onDisconnected = BroadcastChannel<NSUUID>(1)
+    internal val onDisconnected: Flow<NSUUID>
+        // todo: Drop `get()` when `_onDisconnected` is a MutableSharedFlow.
+        get() = _onDisconnected.openSubscription().consumeAsFlow()
+
+    private val _state = MutableStateFlow<CBManagerState?>(null)
+    val state: Flow<CBManagerState> = _state.filterNotNull()
+
+    sealed class Response {
+
+        data class DidDiscoverPeripheral(
+            val cbPeripheral: CBPeripheral,
+            val rssi: NSNumber,
+            val advertisementData: Map<String, Any>,
+        ) : Response()
+    }
+
+    // todo: MutableSharedFlow when Coroutines 1.4.x-mt is released.
+    private val _response = BroadcastChannel<Response>(BUFFERED)
+    val response: Flow<Response> = _response.asFlow()
+
+    sealed class ConnectionEvent {
+
+        abstract val identifier: NSUUID
+
+        data class DidConnect(
+            override val identifier: NSUUID,
+        ) : ConnectionEvent()
+
+        data class DidFailToConnect(
+            override val identifier: NSUUID,
+            val error: NSError?,
+        ) : ConnectionEvent()
+
+        data class DidDisconnect(
+            override val identifier: NSUUID,
+            val error: NSError?,
+        ) : ConnectionEvent()
+    }
+
+    private val _connectionState = MutableStateFlow<ConnectionEvent?>(null)
+    val connectionState: Flow<ConnectionEvent> = _connectionState.filterNotNull()
+
+    /* Monitoring Connections with Peripherals */
+
+    override fun centralManager(
+        central: CBCentralManager,
+        didConnectPeripheral: CBPeripheral,
+    ) {
+        println("<- CentralManagerDelegate didConnect")
+        _connectionState.value = DidConnect(didConnectPeripheral.identifier)
+    }
+
+    @Suppress("CONFLICTING_OVERLOADS") // https://kotlinlang.org/docs/reference/native/objc_interop.html#subclassing-swiftobjective-c-classes-and-protocols-from-kotlin
+    override fun centralManager(
+        central: CBCentralManager,
+        didDisconnectPeripheral: CBPeripheral,
+        error: NSError?,
+    ) {
+        println("<- CentralManagerDelegate didDisconnectPeripheral")
+        _onDisconnected.sendBlocking(didDisconnectPeripheral.identifier) // Used to notify `Peripheral` of disconnect.
+        _connectionState.value = DidDisconnect(didDisconnectPeripheral.identifier, error)
+    }
+
+    @Suppress("CONFLICTING_OVERLOADS") // https://kotlinlang.org/docs/reference/native/objc_interop.html#subclassing-swiftobjective-c-classes-and-protocols-from-kotlin
+    override fun centralManager(
+        central: CBCentralManager,
+        didFailToConnectPeripheral: CBPeripheral,
+        error: NSError?,
+    ) {
+        println("<- CentralManagerDelegate didFailToConnectPeripheral")
+        _connectionState.value = DidFailToConnect(didFailToConnectPeripheral.identifier, error)
+    }
```

Comment:
I read the link but I'm still confused on this. Does Swift allow differentiating overloads by only the parameter names?

Reply:
When targeting Apple targets, yes, that is my understanding. You are overloading based on parameter names.

Reply:
I am _incredibly impressed_ that Kotlin is able to successfully interop with this. It's stuff like that which kills a lot of inter-language compatibility.